torch/_refs/__init__.py

```diff
@@ -1067,6 +1072,46 @@ def amax(
     )
 
 
+@register_decomposition(torch.ops.aten.addr)
+@out_wrapper
+@elementwise_type_promotion_wrapper(
+    type_promoting_args=("self", "vec1", "vec2"),
+    type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.NO_OPMATH,
+)
+def addr(
+    self: TensorLikeType,
+    vec1: TensorLikeType,
+    vec2: TensorLikeType,
+    beta: NumberType = 1,
+    alpha: NumberType = 1,
+) -> TensorLikeType:
+    check(
+        vec1.ndim == 1,
+        lambda: f"addr: Expected 1-D argument vec1, but got {vec1.ndim}-D",
+    )
+    check(
+        vec2.ndim == 1,
+        lambda: f"addr: Expected 1-D argument vec2, but got {vec2.ndim}-D",
+    )
+    self = self.expand(vec1.shape[0], vec2.shape[0])
+    check(
+        self.ndim == 2, lambda: f"2D tensor expected, got {self.ndim}D tensor for input"
+    )
+    check(
+        self.shape[0] == vec1.shape[0] and self.shape[1] == vec2.shape[0],
```

Comment:
Won't the expand have already asserted this?

Reply:
Once again, faithful representation of the original code :) My rule for porting: first do a slavish port, and then once that passes all tests, then refactor.

Reply:
haha, OK I guess -- but some cleanups seem easy to snipe